unit/analyses/ai/ai.cpp

```diff
@@ -0,0 +1,310 @@
+  // h() { int hy = 0; }
+
+  register_language(new_ansi_c_language);
+  config.ansi_c.set_LP64();
+
+  goto_modelt goto_model;
+
+  // g:
+
+  symbolt gy;
+  gy.name = "gy";
+  gy.mode = ID_C;
+  gy.type = signed_int_type();
+
+  symbolt g;
+  g.name = "g";
+  g.mode = ID_C;
+  g.type = code_typet({}, void_typet());
+  g.value = code_assignt(gy.symbol_expr(), from_integer(0, signed_int_type()));
+
+  // h:
+
+  symbolt hy;
+  hy.name = "hy";
+  hy.mode = ID_C;
+  hy.type = signed_int_type();
+
+  symbolt h;
+  h.name = "h";
+  h.mode = ID_C;
+  h.type = code_typet({}, void_typet());
+  h.value = code_assignt(hy.symbol_expr(), from_integer(0, signed_int_type()));
+
+  goto_model.symbol_table.add(g);
+  goto_model.symbol_table.add(gy);
+  goto_model.symbol_table.add(h);
+  goto_model.symbol_table.add(hy);
+
+  // f:
+
+  symbolt x;
+  x.name = "x";
+  x.mode = ID_C;
+  x.type = signed_int_type();
+
+  symbolt y;
+  y.name = "y";
+  y.mode = ID_C;
+  y.type = signed_int_type();
+
+  goto_model.symbol_table.add(x);
+  goto_model.symbol_table.add(y);
+
+  code_blockt f_body;
+
+  f_body.copy_to_operands(code_declt(x.symbol_expr()));
+  f_body.copy_to_operands(code_declt(y.symbol_expr()));
+
+  f_body.copy_to_operands(
+    code_assignt(x.symbol_expr(), from_integer(10, signed_int_type())));
+  f_body.copy_to_operands(
+    code_assignt(y.symbol_expr(), from_integer(20, signed_int_type())));
+
+  f_body.copy_to_operands(make_void_call(h.symbol_expr()));
+
+  code_whilet loop;
+
+  loop.cond() =
+    notequal_exprt(x.symbol_expr(), from_integer(0, signed_int_type()));
+
+  code_blockt loop_body;
+  loop_body.copy_to_operands(
+    code_assignt(
+      x.symbol_expr(),
+      minus_exprt(x.symbol_expr(), from_integer(1, signed_int_type()))));
+  loop_body.copy_to_operands(
+    code_assignt(
+      y.symbol_expr(),
+      minus_exprt(y.symbol_expr(), from_integer(1, signed_int_type()))));
+  loop_body.copy_to_operands(make_void_call(g.symbol_expr()));
+
+  loop.body() = loop_body;
+
+  f_body.move_to_operands(loop);
+
+  f_body.copy_to_operands(make_void_call(g.symbol_expr()));
+
+  symbolt f;
+  f.name = "f";
+  f.mode = ID_C;
+  f.type = code_typet({}, void_typet());
+  f.value = f_body;
+
+  goto_model.symbol_table.add(f);
+
+  // __CPROVER_start:
+
+  symbolt start;
+  start.name = goto_functionst::entry_point();
+  start.mode = ID_C;
+  start.type = code_typet({}, void_typet());
+  start.value = make_void_call(f.symbol_expr());
+
+  goto_model.symbol_table.add(start);
+
+  null_message_handlert nullout;
+  goto_convert(goto_model, nullout);
+
+  WHEN("The target program is analysed")
+  {
+    instruction_counter_analysist example_analysis;
+    example_analysis(goto_model);
+
+    THEN("No state should be bottom")
+    {
+      forall_goto_functions(f_it, goto_model.goto_functions)
+      {
+        forall_goto_program_instructions(i_it, f_it->second.body)
+        {
+          REQUIRE(!example_analysis[i_it].is_bottom());
```

Comment:
`REQUIRE_FALSE` is slightly easier to understand (same comment applies to all other uses of `REQUIRE(!something)`